db/migrate/20181129000515_create_attributes_table.rb

```diff
@@ -0,0 +1,27 @@
+class CreateAttributesTable < ActiveRecord::Migration[5.1]
+  def change
+    this_hostid = Machine.new.hostid
+
+    create_table :attributes, id: :bigint do |t|
+      t.string :container_type, null: false
+      t.integer :container_id, limit: 8, null: false
+      t.jsonb :value, null: false, default: '{}'
```

Comment:
Would it make sense to constrain `value` in some way?  I assume this is just supposed to be a collection of key-value pairs.

Reply:
Yes, but really _any_ JSON document?

Reply:
yup, i don't want to limit this table. The whole idea of this table is for people to add any data to any existing row in the database.

Reply:
So even though we call this table "attributes", the json really could be anything, it doesn't even have to be a hash, yes?

Reply:
it has to be json format that is all we know. We don't know what it will contain.

Reply:
OK, I'll just assume you know what you're doing.